test/test_dbc.cpp

```diff
@@ -49,4 +49,80 @@ TEST_CASE("Testing dbc file loading", "[fileio]") {
 		REQUIRE(parser->get_messages().front().signals == msg.signals);
 	}
 
-}
\ No newline at end of file
+}
+
+/*!
+ * \brief TEST_CASE
+ * Test negative values in offset, min, max
+ */
+TEST_CASE("Testing negative values") {
+  const auto* filename = std::tmpnam(NULL);
+
+  auto* file = std::fopen(filename, "w");
+  CHECK(file);
+
+  std::fputs(PRIMITIVE_DBC.c_str(), file);
+  std::fputs(R"(BO_ 234 MSG1: 8 Vector__XXX
+ SG_ Sig1 : 55|16@0- (0.1,0) [-3276.8|-3276.7] "C" Vector__XXX
+ SG_ Sig2 : 39|16@0- (0.1,0) [-3276.8|-3276.7] "C" Vector__XXX
+ SG_ Sig3 : 23|16@0- (10,0) [-3276.8|-3276.7] "C" Vector__XXX
+ SG_ Sig4 : 7|16@0- (1,-10) [0|32767] "" Vector__XXX)", file);
+  std::fclose(file);

```

Comment:
One quick refactor I would suggest is to pull this out into a function `void create_tmp_dbc_with(const std::string_view content)`at the top of the test file here. That way you can pull out the duplicate code in the case below

Reply:
I would preferr not doing this, because consider you are adding additional tests then you have to create another function. I also don't like copy paste, but in tests I prefer, so when I change it at one position I don't break all tests.

Reply:
I guess why would you need another function in that case? The only differences are the content you are putting in the tmp file right so the two function calls would like such:
```c++
create_tmp_dbc_with(R"(BO_ 234 MSG1: 8 Vector__XXX
 SG_ Sig1 : 55|16@0- (0.1,0) [-3276.8|-3276.7] "C" Vector__XXX
 SG_ Sig2 : 39|16@0- (0.1,0) [-3276.8|-3276.7] "C" Vector__XXX
 SG_ Sig3 : 23|16@0- (10,0) [-3276.8|-3276.7] "C" Vector__XXX
 SG_ Sig4 : 7|16@0- (1,-10) [0|32767] "" Vector__XXX)"
);

// In the other test case
create_tmp_dbc_with(R"(BO_ 234 MSG1: 8 Vector__XXX
 SG_ Speed : 0|8@1+ (1,0) [0|204] "Km/h"  DEVICE1,DEVICE2,DEVICE3)"
);
```

Reply:
The actual function would be:
```c++
  auto* file = std::fopen(filename, "w");
  CHECK(file);

  std::fputs(PRIMITIVE_DBC.c_str(), file);
  std::fputs(content, file);
  std::fclose(file);
```

Reply:
Ah ok I understand yes this would be possible

Reply:
Is it fine using std::string instead of string_view, because string_view requires c++17 and labplot currently uses c++11.

Reply:
changed in a666b5260d1bd8fc2ff32836e202cee06965ddd3

Reply:
yea that is fine with just `const char *`. I forgot that i had it set to c++11. I left a comment about the inclusion of the `<string_view>` header. Once that is removed I am good to merge.

Reply:
I removed it in 2a4d0598f28aefa2c1218aa363b1ff7aac6e2ff3